ui/dashboards/src/components/PanelDrawer/PanelEditorForm.tsx

```diff
@@ -0,0 +1,111 @@
+
+  const { layouts } = useLayouts();
+
+  const [name, setName] = useState(initialValues.name);
+  const [description, setDescription] = useState(initialValues.description);
+  const [group, setGroup] = useState(initialValues.group);
+  const [kind, setKind] = useState(initialValues.kind);
+  const { spec, onSpecChange } = usePanelSpecState(kind, initialValues.spec);
+
+  // Ignore string values (which would be an "empty" value from the Select) since we don't allow them to unset it
+  const handleGroupChange: SelectProps<number>['onChange'] = (e) => {
+    const { value } = e.target;
+    if (typeof value === 'string') {
+      return;
+    }
+    setGroup(value);
+  };
+
+  const handleSubmit: FormEventHandler = (e) => {
+    e.preventDefault();
+    const values: PanelEditorFormValues = { name, description, group, kind, spec };
+    onSubmit(values);
+  };
+
+  return (
+    <form id={panelEditorFormId} onSubmit={handleSubmit}>
+      <Grid container spacing={2}>
+        <Grid item xs={8}>
+          <TextField
+            required
+            label="Panel Name"
+            value={name}
+            variant="outlined"
+            onChange={(e) => setName(e.target.value)}
+          />
+        </Grid>
+        <Grid item xs={4}>
+          <FormControl>
+            <InputLabel id="select-group">Group</InputLabel>
+            <Select required labelId="select-group" label="Group" value={group ?? 0} onChange={handleGroupChange}>
+              {layouts.map((layout, index) => (
+                <MenuItem key={index} value={index}>
+                  {layout.spec.display?.title ?? `Group ${index + 1}`}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid>
+        <Grid item xs={8}>
+          <TextField
+            label="Panel Description"
+            value={description}
+            variant="outlined"
+            onChange={(e) => setDescription(e.target.value)}
+          />
+        </Grid>
+        <Grid item xs={4}>
+          <FormControl>
+            <InputLabel id="panel-type-label">Panel Type</InputLabel>
+            <PanelTypeSelect
+              required
+              labelId="panel-type-label"
+              label="Panel Type"
+              value={kind}
+              onChange={(e) => setKind(e.target.value)}
+            />
+          </FormControl>
+        </Grid>
+        <Grid item xs={12}>
+          <ErrorBoundary FallbackComponent={ErrorAlert}>
+            {/* Wait until we have some proper initial spec values before rendering the editor */}
+            {spec !== undefined && <PanelSpecEditor panelPluginKind={kind} value={spec} onChange={onSpecChange} />}
+          </ErrorBoundary>
+        </Grid>
+      </Grid>
+    </form>
+  );
+}
+
+/**
+ * The `id` attribute added to the `PanelEditorForm` component, allowing submit buttons to live outside the form.
+ */
+export const panelEditorFormId = 'panel-editor-form';
```

Comment:
Something I learned while working on this PR is that you can have submit buttons outside of the `form` element with just some vanilla HTML props. 🧐